### DISCUSSION TITLE
Can't guess your region for bucket problem

### DISCUSSION THREAD

**Discussion by jueshali** [juicedata/juicefs#127] [Q&A] [answered]
When I tried to connect to s3, the following error occurred, and I didn't find that I could set the region parameters. Am I missing something? Following is my status code. Thanks a lot for help.
Object storage: Can't guess your region for bucket : AccessDenied: Access Denied
        status code: 403, request id: CJ0Q3ZFSEX1W6K5Y, host id: j1w7L/6CUUxQ11OI6

**Comment by xiaogaozi** [comment, marked as answer]
Can you provide the complete `juicefs format` command? Looks like the `--bucket` option is incorrect.

**Reply by jueshali** [reply]
./juicefs format --storage s3   --bucket s3://    --access-key ****************   --secret-key ****************  localhost gdc-tokyo-test

**Reply by jueshali** [reply]
My S3 storage address is tokyo, and I changed the system variable setting echo $AWS_DEFAULT_REGION = ap-northeast-1，but it doesn't work @xiaogaozi

**Reply by xiaogaozi** [reply]
Currently, JuiceFS only support [virtual hosted-style URI](https://docs.aws.amazon.com/AmazonS3/latest/dev/VirtualHosting.html). So you need specify `--bucket` looks like this:

```
$ ./juicefs format --storage s3 --bucket https://<bucket>.s3.<region>.amazonaws.com --access-key **************** --secret-key **************** localhost gdc-tokyo-test
```

Replace `<bucket>` with your bucket name and for Tokyo the region is `ap-northeast-1`.

**Reply by jueshali** [reply]
Yes, thanks a lot. I found the relevant code, I hope this can help other people who use this software and encounter similar problems
```golang
if strings.Contains(uri.Host, ".amazonaws.com") {
			// standard s3
			// [BUCKET].s3-[REGION].[REST_OF_ENDPOINT]
			// [BUCKET].s3.[REGION].amazonaws.com[.cn]
			hostParts = strings.SplitN(uri.Host, ".s3", 2)
```

**Reply by xiaogaozi** [reply]
Thank you for your feedback. We'll document this as soon as possible. The related issue is #96.

**Reply by xiaogaozi** [reply]
The document is online: https://github.com/juicedata/juicefs/blob/main/docs/how_to_setup_object_storage.md
